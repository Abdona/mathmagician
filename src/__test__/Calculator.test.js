import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import MyCalculator from '../components/Calculator';
import '@testing-library/jest-dom';

describe('Test MyCalculator Components', () => {
  it('test MyCalculator function component renders without error', () => {
    const divElement = document.createElement('div');
    ReactDOM.render(<MyCalculator />, divElement);
  });

  it('render correctly', () => {
    const { getByTestId } = render(<MyCalculator />);
    expect(getByTestId('mycalc')).toHaveTextContent('Let`s do some mathAC+/-%รท789x456-123+0.=');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<MyCalculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test user interaction simultaion', () => {
  it('test adding two numbers ', () => {
    render(<MyCalculator />);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByDisplayValue('')).toContainHTML('5');
  });

  // 5- 3 = 2
  it('test subtracting two numbers ', () => {
    render(<MyCalculator />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('-'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByDisplayValue('')).toContainHTML('2');
  });

  // 10 x 3 = 30
  it('test Multiplying two numbers ', () => {
    render(<MyCalculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByDisplayValue('')).toContainHTML('30');
  });

  it('test Dividing two numbers ', () => {
    render(<MyCalculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('รท'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByDisplayValue('')).toContainHTML('5');
  });

  it('test Mod Operation two numbers ', () => {
    render(<MyCalculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('%'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByDisplayValue('')).toContainHTML('1');
  });
});
