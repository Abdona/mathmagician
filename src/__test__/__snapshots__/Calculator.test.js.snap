// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyCalculator Components matches snapshot 1`] = `
<div
  className="page-section"
  data-testid="mycalc"
>
  <h1>
    Let\`s do some math
  </h1>
  <div
    className="grid-continer"
  >
    <input
      className="resultscreen"
      placeholder={0}
      readOnly={true}
    />
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="buttoperation"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="buttoperation"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="buttoperation"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="buttoperation"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zeroitem"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="buttoperation"
      id="eqoperation"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
